browser/ui/webui/brave_rewards_ui.cc

```diff
@@ -893,45 +888,50 @@ void RewardsDOMHandler::CheckImported(const base::ListValue *args) {
 }
 
 void RewardsDOMHandler::GetAdsData(const base::ListValue *args) {
-  if (ads_service_ && web_ui()->CanCallJavascript()) {
-    base::DictionaryValue adsData;
+  if (!ads_service_ || !web_ui()->CanCallJavascript()) {
+    return;
+  }
 
-    bool ads_ui_enabled;
-    bool ads_enabled = ads_service_->is_enabled();
-    int ads_per_hour = ads_service_->ads_per_hour();
+  base::DictionaryValue ads_data;
 
-    #if BUILDFLAG(BRAVE_ADS_ENABLED)
-      ads_ui_enabled = true;
-    #else
-      ads_ui_enabled = false;
-    #endif
+  auto is_supported_region = ads_service_->IsSupportedRegion();
+  ads_data.SetBoolean("adsIsSupported", is_supported_region);
 
-    adsData.SetBoolean("adsEnabled", ads_enabled);
-    adsData.SetInteger("adsPerHour", ads_per_hour);
-    adsData.SetBoolean("adsUIEnabled", ads_ui_enabled);
+  auto is_ads_enabled = ads_service_->IsAdsEnabled();
+  ads_data.SetBoolean("adsEnabled", is_ads_enabled);
 
-    web_ui()->CallJavascriptFunctionUnsafe("brave_rewards.adsData", adsData);
-  }
+  auto ads_per_hour = ads_service_->GetAdsPerHour();
+  ads_data.SetInteger("adsPerHour", ads_per_hour);
+
+  #if BUILDFLAG(BRAVE_ADS_ENABLED)
+    auto ads_ui_enabled = true;
+  #else
+    auto ads_ui_enabled = false;
+  #endif
+  ads_data.SetBoolean("adsUIEnabled", ads_ui_enabled);
+
+  web_ui()->CallJavascriptFunctionUnsafe("brave_rewards.adsData", ads_data);
 }
 
 void RewardsDOMHandler::SaveAdsSetting(const base::ListValue* args) {
-  if (ads_service_) {
-    std::string key;
-    std::string value;
-    args->GetString(0, &key);
-    args->GetString(1, &value);
+  if (!ads_service_) {
+    return;
+  }
 
-    if (key == "adsEnabled") {
-      ads_service_->set_ads_enabled(value == "true");
-    }
+  std::string key;
+  args->GetString(0, &key);
 
-    if (key == "adsPerHour") {
-      ads_service_->set_ads_per_hour(std::stoi(value));
-    }
+  std::string value;
+  args->GetString(1, &value);
 
-    base::ListValue* emptyArgs;
-    GetAdsData(emptyArgs);
+  if (key == "adsEnabled") {
+    ads_service_->SetAdsEnabled(value == "true");
+  } else if (key == "adsPerHour") {
+    ads_service_->SetAdsPerHour(std::stoull(value));
```

Comment:
Seems weird that we read these two values as strings and then convert them to other types. Is it not possible to send them as the correct types in the first place?

Reply:
@ryanml ^^